### PR TITLE
Memory histogram

### PR DESCRIPTION
**Pull request by mdboom**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `HISTORY.rst` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean_parser/schemas/metrics.1-0-0.schema.yaml [hunk 1]
```diff
@@ -294,6 +298,24 @@ definitions:
           - hour
           - day
 
+      memory_unit:
+        title: Memory unit
+        description: |
+          The unit that the incoming memory size values are recorded in.
+
+          The units are the power-of-2 units, so "kilobyte" is pedantically a "kibibyte".
```
Comment (chutten) [inline glean_parser/schemas/metrics.1-0-0.schema.yaml]: Well actually a more accurate adjective would be "correctly". \*pushes bridge of glasses up nose*

File: glean_parser/schemas/metrics.1-0-0.schema.yaml [hunk 2]
```diff
@@ -294,6 +298,24 @@ definitions:
...
+
+          - kilobyte == 2^10 ==         1,024 bytes
+          - megabyte == 2^20 ==     1,048,576 bytes
+          - gigabyte == 2^30 == 1,073,741,824 bytes
+
```
Comment (chutten) [inline glean_parser/schemas/metrics.1-0-0.schema.yaml]: Include a note about truncation/rounding? (Even if there is none)

File: glean_parser/schemas/metrics.1-0-0.schema.yaml (hunk 1 above)
Comment (mdboom) [reply glean_parser/schemas/metrics.1-0-0.schema.yaml]: Indeed.  That's probably a better (and less condescending) word to use.

File: glean_parser/schemas/metrics.1-0-0.schema.yaml (hunk 2 above)
Comment (mdboom) [reply glean_parser/schemas/metrics.1-0-0.schema.yaml]: Good point.

File: glean_parser/schemas/metrics.1-0-0.schema.yaml (hunk 1 above)
Comment (chutten) [reply glean_parser/schemas/metrics.1-0-0.schema.yaml]: And here I was saying it as a joke.

File: glean_parser/schemas/metrics.1-0-0.schema.yaml [hunk 3]
```diff
@@ -450,7 +474,11 @@ additionalProperties:
                 enum:
                   - timing_distribution
                   - custom_distribution
+                  - memory_distribution
```
Comment (Dexterp37) [inline glean_parser/schemas/metrics.1-0-0.schema.yaml]: Should we update `test_geckoview_only_on_valid_metrics` to cover this too?

**Pull request merged**